include/otbObiaGenericRegionMergingFilter.txx

```diff
@@ -160,12 +160,14 @@ GenericRegionMergingFilter<TInputGraph, TOutputGraph, TMergingCostFunc, THeurist
 {
     std::cout << "Compute Merging cost" << std::endl;
     // Retrieve the output graph.
+
     auto outputGraph = this->GetOutput();
 
     MergingCostValueType minCost;
     // Fix : uninitialized variable
     uint64_t minNodeId = outputGraph->GetNumberOfNodes()+1, idx, minIdx;
 
+    std::cout << "updating attribute cost updated ...";
```

Comment:
There is a new behavior in otb regarding logs, see : https://gitlab.orfeo-toolbox.org/orfeotoolbox/otb/merge_requests/20

You could use `otbLogMacro(Debug,"Compute Merging cost");` instead, and enable/disable debug log using the `OTB_LOGGER_LEVEL` environment variable.